src/sync/sys/windows/net.rs

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) 2019 Ant Financial
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::error::Result;
+
+use mio::windows::NamedPipe;
+
+use std::ffi::OsStr;
+use std::fs::OpenOptions;
+use std::os::windows::ffi::OsStrExt;
+use std::os::windows::fs::OpenOptionsExt;
+use std::os::windows::io::{FromRawHandle, IntoRawHandle, RawHandle};
+use std::os::windows::prelude::AsRawHandle;
+use std::sync::atomic::{AtomicBool, AtomicI32, Ordering};
+use std::sync::{Arc, Mutex};
+use std::{io};
+
+
+use windows_sys::Win32::Foundation::{ERROR_NO_DATA, INVALID_HANDLE_VALUE, CloseHandle, GetLastError};
+use windows_sys::Win32::Storage::FileSystem::{
+    FILE_FLAG_FIRST_PIPE_INSTANCE, FILE_FLAG_OVERLAPPED, PIPE_ACCESS_DUPLEX,
+};
+use windows_sys::Win32::System::Pipes::{
+    CreateNamedPipeW, PIPE_TYPE_BYTE, PIPE_UNLIMITED_INSTANCES,
+};
+use mio::{Events, Interest, Poll, Token};
+use std::io::{Read, Write};
+
+const SERVER: Token = Token(0);
+const CLIENT: Token = Token(1);
+
+pub struct PipeListener {
+    first_instance: AtomicBool,
+    address: String,
+    instance_number: AtomicI32,
+}
+
+impl PipeListener {
+    pub(crate) fn new(sockaddr: &str) -> Result<PipeListener> {
+        Ok(PipeListener {
+            first_instance: AtomicBool::new(true),
+            address: sockaddr.to_string(),
+            instance_number: AtomicI32::new(1),
+        })
+    }
+
+    pub(crate) fn accept(
+        &self,
+        quit_flag: &Arc<AtomicBool>,
+    ) -> std::result::Result<Option<PipeConnection>, io::Error> {
+        if quit_flag.load(Ordering::SeqCst) {
+            info!("listener shutdown for quit flag");
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                "listener shutdown for quit flag",
+            ));
+        }
+
+        // Create a new pipe for every new client
+        let mut namedpipe = self.new_instance().unwrap();
+
+        let mut poll = Poll::new().unwrap();
+        let mut events = Events::with_capacity(1024);
+        poll.registry()
+            .register(
+                &mut namedpipe,
+                SERVER,
+                Interest::WRITABLE,
+            )
+            .unwrap();
```

Comment:
Is it safe to unwrap? You can return Err here just in case.

Reply:
removed mio so this isn't applicable anymore